### DISCUSSION TITLE
Vulnerabilities should be associated with the container images and not Kubernetes resource

### DISCUSSION THREAD

**Discussion by nilesh-akhade** [aquasecurity/trivy#3198] [Q&A]
The question is about Trivy CLI. The scan result from the _trivy kubernetes_ command shows duplicates vulnerability results if the same container image is being used in multiple Kubernetes resources.
 
We deploy two deployments in the customer's cluster both use the same container image. When the customer's cluster is scanned using trivy, the report contains the same vulnerability report twice.

Do you agree with the statement, vulnerabilities should be associated with the container images and not the Kubernetes resources?

--
Nilesh Akhade
_Developer at Catalogic_

**Comment by knqyf263** [comment]
@chen-keinan What do you think?

**Comment by chen-keinan** [comment]
@nilesh-akhade definitely vulnerabilities should be associated with the container images level.
I assume it can be improved to consolidate similar kubernetes resource and presenting vulnerabilities only once.
do you mind opening issue for it

**Comment by nilesh-akhade** [comment]
https://github.com/aquasecurity/trivy/issues/3220
